src/ideal/ideal.jl

```diff
@@ -495,16 +495,8 @@ function sres(I::sideal{T}, max_length::Int) where T <: Nemo.RingElem
         max_length = nvars(R)
         # TODO: consider qrings
    end
-   r, length, minimal = libSingular.id_sres(I.ptr, Cint(max_length + 1), R.ptr)
-   for i = 1:length
-      ptr = libSingular.getindex(r, Cint(i - 1))
-      if ptr.cpp_object == C_NULL
-         length = i - 1
-         break
-      end
-     libSingular.idSkipZeroes(ptr)
-   end
-   return sresolution{T}(R, length, r, minimal)
+   r, minimal = libSingular.id_sres(I.ptr, Cint(max_length + 1), R.ptr)
+   return sresolution{T}(R, r, minimal)
```

Comment:
I am wondering: in how far are this change and the many others similar ones tied to our library calling work? Perhaps they could / should be separated? (Just wondering -- of course that is extra work, and so we may not want it; the main reason I bring it up is that this PR is huuuge and it's typically easier to review, test and merge multiple smaller PRs, as you of course know -- it's a balance between your time, and that of reviewers :-)

Reply:
Let's just get the whole thing merged. Don't bother splitting it.